homeassistant/components/omnilogic/switch.py

```diff
@@ -0,0 +1,264 @@
+"""Platform for Omnilogic switch integration."""
+import time
+
+from omnilogic import OmniLogicException
+import voluptuous as vol
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.helpers import config_validation as cv, entity_platform
+
+from .common import OmniLogicEntity, OmniLogicUpdateCoordinator, check_guard
+from .const import COORDINATOR, DOMAIN, PUMP_TYPES
+
+SERVICE_SET_SPEED = "set_pump_speed"
+OMNILOGIC_SWITCH_OFF = 7
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the light platform."""
+
+    coordinator = hass.data[DOMAIN][entry.entry_id][COORDINATOR]
+    entities = []
+
+    for item_id, item in coordinator.data.items():
+        id_len = len(item_id)
+        item_kind = item_id[-2]
+        entity_settings = SWITCH_TYPES.get((id_len, item_kind))
+
+        if not entity_settings:
+            continue
+
+        for entity_setting in entity_settings:
+            for state_key, entity_class in entity_setting["entity_classes"].items():
+                if check_guard(state_key, item, entity_setting):
+                    continue
+
+                entity = entity_class(
+                    coordinator=coordinator,
+                    state_key=state_key,
+                    name=entity_setting["name"],
+                    kind=entity_setting["kind"],
+                    item_id=item_id,
+                    icon=entity_setting["icon"],
+                )
+
+                entities.append(entity)
+
+    async_add_entities(entities)
+
+    # register service
+    platform = entity_platform.current_platform.get()
+
+    platform.async_register_entity_service(
+        SERVICE_SET_SPEED,
+        {vol.Required("speed"): cv.positive_int},
+        "async_set_speed",
+    )
+
+
+class OmniLogicSwitch(OmniLogicEntity, SwitchEntity):
+    """Define an Omnilogic Base Switch entity which will be instantiated through specific switch type entities."""
```

Comment:
Please write a shorter docstring header. Docstrings should normally be under 72 characters per line. If more text is required it can be in the docstring body. The docstring header and body should be separated by a blank line.